### ISSUE TITLE
Rstudio-2021.09.2-382 version, dir() can't display Chinese characters normally.

### ISSUE THREAD

**Issue by wangbinzjcc** [rstudio/rstudio#10451]
### System details

    RStudio Edition :  Desktop
    RStudio Version :  RStudio-2021.09.2-382
    OS Version      :  Windows 10 x64
    R Version       :  R version 4.1.2

### Steps to reproduce the problem

> dir("D:/test", full.names = T)
[1] "D:/test/涓枃"     "D:/test/涓枃.txt"

### Describe the problem in detail

In the version of RStudio-2021.09.1-372, the file path with Chinese characters can be displayed normally by using dir () in the console.
However, In the versions of RStudio-2021.09.2-382 and RStudio-2022.02.0-preview-392,  the file path with Chinese characters can't be displayed normally by using dir () in the console.

[dir_chinese_path.txt](https://github.com/rstudio/rstudio/files/7946909/dir_chinese_path.txt)

### Describe the behavior you expected

dir("D:/test", full.names = T)
[1] "D:/test/中文" "D:/test/中文.txt"

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report. I'm not entirely sure what's going on, but I'm 99% sure it'll be fixed when R 4.2.0 is released, with UTF-8 locale support.

**Comment by kevinushey** [comment]
As a workaround, you can do something like:

```
files <- dir("D:/test", full.names = TRUE)
Encoding(files) <- "UTF-8"
files
```

and you should hopefully see the correct Chinese text.

**Comment by wangbinzjcc** [comment]
@kevinushey Thank you. The workaround method is feasible. Look forward to the arrival of version R 4.2.0.

**Comment by wangbinzjcc** [comment]
The temporary solution still can't solve the Chinese path problem.

For example: dir("D:/test/中文", full.names = TRUE)

Run directly in R 4.1.2, or use Rstudio-2021.09.1-372 version,  the correct results can be obtained.
However, using Rstudio-2021.09.2-382 and above, the correct results cannot be obtained.

I had to temporarily return to the old version of Rstudio-2021.09.1-372.


![R](https://user-images.githubusercontent.com/4001597/151331009-498973f7-4d3d-47e1-b8f3-6f6e1ebe6db8.png)
![RStudio](https://user-images.githubusercontent.com/4001597/151331020-fe749783-4150-45d8-b41a-1a172227ae74.png)

**Comment by kevinushey** [comment]
Can you confirm in RStudio what locale you have active? Are you working within a Chinese locale, or English locale? (What is the output of `Sys.getlocale()`)?

**Comment by wangbinzjcc** [comment]
![sys getlocale](https://user-images.githubusercontent.com/4001597/151471298-15945a93-a237-428b-ba7f-824efdac48ac.png)

**Comment by ronblum** [comment]
I can reproduce the problem in Ghost Orchid 2021.09.2+382 on Windows 10 with R 4.1.2, and it works with R 4.2.0. 

In Prairie Trillium 2022.02.0+418 on Windows 10 with R 4.1.2, and setting the locale values to `Chinese (Simplified)_China.936`, I get Chinese characters. However, they're different from the file and directory names:
> \> dir("C:/test", full.names = TRUE)
[1] "C:/test/涓枃"     "C:/test/涓枃.txt"

**Comment by kevinushey** [comment]
Another example for testing:

```

tdir <- tempfile()
dir.create(tdir)
owd <- setwd(tdir)

dir.create("你好")
file.create("世界")

Sys.setlocale(locale = "English")
list.files()

Sys.setlocale(locale = "Chinese")
list.files()
```

You should see:

```
> list.files()
[1] "你好" "世界"
```

regardless of the locale.

**Comment by gsmolinski** [comment]
Don't know, how obvious is this for you, but the same problem is with Polish locale.

```
tdir <- tempfile()
dir.create(tdir)
owd <- setwd(tdir)

dir.create("ń") # right alt + n 

Sys.setlocale(locale = "Polish")
dir()

[1] "Ĺ„"
```

**Comment by ronblum** [comment]
Confirmed fixed in RStudio Desktop 2022.02.0+421  on Windows 10 with R 4.1.0 and R 4.2.0 (devel). It's possible I wasn't setting the locale properly the first time around.

```
> # Chinese (original post)
> Sys.setlocale(locale = "Chinese")
[1] "LC_COLLATE=Chinese (Simplified)_China.936;LC_CTYPE=Chinese (Simplified)_China.936;LC_MONETARY=Chinese (Simplified)_China.936;LC_NUMERIC=C;LC_TIME=Chinese (Simplified)_China.936"
> dir("c:/test", full.names = T)
[1] "c:/test/中文"     "c:/test/中文.txt"
> # Chinese (temp files)
> tdir <- tempfile()
> dir.create(tdir)
> owd <- setwd(tdir)
> dir.create("你好")
> file.create("世界")
[1] TRUE
> Sys.setlocale(locale = "English")
[1] "LC_COLLATE=English_United States.1252;LC_CTYPE=English_United States.1252;LC_MONETARY=English_United States.1252;LC_NUMERIC=C;LC_TIME=English_United States.1252"
> list.files()
[1] "世界" "你好"
> Sys.setlocale(locale = "Chinese")
[1] "LC_COLLATE=Chinese (Simplified)_China.936;LC_CTYPE=Chinese (Simplified)_China.936;LC_MONETARY=Chinese (Simplified)_China.936;LC_NUMERIC=C;LC_TIME=Chinese (Simplified)_China.936"
> list.files()
[1] "你好" "世界"
> # Polish temp directory
> tdir <- tempfile()
> dir.create(tdir)
> owd <- setwd(tdir)
> Sys.setlocale(locale = "Polish")
[1] "LC_COLLATE=Polish_Poland.1250;LC_CTYPE=Polish_Poland.1250;LC_MONETARY=Polish_Poland.1250;LC_NUMERIC=C;LC_TIME=Polish_Poland.1250"
> dir.create("ń") # right alt + n
> dir()
[1] "ń"
> Sys.setlocale(locale = "English")
[1] "LC_COLLATE=English_United States.1252;LC_CTYPE=English_United States.1252;LC_MONETARY=English_United States.1252;LC_NUMERIC=C;LC_TIME=English_United States.1252"
> list.files()
[1] "ń"
```
